app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -500,8 +541,9 @@ private void queueDownload(Download download) {
 
         request.allowScanningByMediaScanner();
 
-        final DownloadManager manager = (DownloadManager) context.getSystemService(Context.DOWNLOAD_SERVICE);
-        manager.enqueue(request);
+        long downloadReference = manager.enqueue(request);
+
+        DownloadBroadcastReceiver.addQueuedDownload(downloadReference);
```

Comment:
Is there any reason for this to be static?

In general, mixing static (these queued downlads) and member variables (our receiver instance) can create bugs since it's harder to create track of the lifecycles of your program state. For example, if it's all static (it lasts basically forever) or member (it lasts as long as the object does) but if they're mixed, it's easy to overlook.

In this specific case, we may overlook the system's lifecycles: would the system return a new downloadManager instance if the Activity (and thus this broadcast receiver) is recreated? If it's a new instance, do the download references within the new instance match the download references we had already stored statically? I'm not sure... it's safer to stick with member variables!

That being said, static *does* let us do more (if the Activity is recreated, we can avoid losing any pending download references as we would with member variables) but there's always a question of code complexity vs. functionality. imo, it's unlikely the Activity to be destroyed so I'm fine dropping download snackbars if it happens (especially because the system sends notifications too) to make the code much less complex.